### PR TITLE
VEN-379 | Customers Page improvements

### PR DESCRIPTION
**Pull request by Nizar-Rahme**

### CODE REVIEW

File: src/domain/customers/__mocks__/data.ts [hunk 1]
```diff
@@ -8,16 +8,15 @@ export const dummyCustomers = {
             contactMethod: 'EMAIL',
             email: 'jussi@example.com',
             firstName: 'Jussi',
-            id:
```
Comment (Zeukkari) [inline src/domain/customers/__mocks__/data.ts]: This mockdata is outdated.

File: src/domain/customers/utils.ts [hunk 2]
```diff
@@ -2,29 +2,22 @@ import { CUSTOMERS } from './__generated__/CUSTOMERS';
...
         };
-        return profileData;
+        return [...acc, profileData];
       }
-      return [];
-    });
-  }
-  return [];
+      return acc;
+    }, []) ?? []
```
Comment (Zeukkari) [inline src/domain/customers/utils.ts]: Is this on purpose?

File: tsconfig.json [hunk 3]
```diff
@@ -1,15 +1,10 @@
...
-    "lib": [
-      "dom",
-      "dom.iterable",
-      "esnext"
-    ],
+    "lib": ["dom", "dom.iterable", "esnext"],
     "allowJs": true,
     "skipLibCheck": true,
     "esModuleInterop": true,
-    "noImplicitAny": false,
```
Comment (Zeukkari) [inline tsconfig.json]: Good catch.

File: src/domain/customers/utils.ts [hunk 4]
```diff
@@ -2,29 +2,22 @@ import { CUSTOMERS } from './__generated__/CUSTOMERS';
 
 export interface CustomerData {
   id: string;
-  firstName: string;
-  lastName: string;
-  nickname: string | null;
-  email: string | false | null;
-  phone: string | false | null;
+  name: string;
```
Comment (Zeukkari) [inline src/domain/customers/utils.ts]: Merging firstName and lastName fields here is fine, but it doesn't make much sense to have a seperate nickname field after that.

File: src/domain/customers/utils.ts (hunk 2 above)
Comment (Zeukkari) [reply src/domain/customers/utils.ts]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator

**Comment by Zeukkari** [comment]
my approach had CustomerData and TableData separate by design because I've been overly fixated on caching.
CustomerData <-> TableData
- Customer Data contains graphql query results
- Table Data contains datatable contents for customers page
- Table Data is mapped from Customer Data
- Additionally, individual customer page data will be mapped from Customer Data

**Comment by Nizar-Rahme** [comment]
Closing in favor of https://github.com/City-of-Helsinki/berth-reservations-admin/pull/78

**Pull request closed without merging**